### DISCUSSION TITLE
Can't get email notifications to work

### DISCUSSION THREAD

**Discussion by tjakubo** [umputun/remark42#1642] [Q&A] [answered]
Hey, this project has been of amazing use to me, but I can't seem to get email comment notifications to work.
I've configured all the relevant options when running remark, my startup script:
```
#!/bin/bash
./remark42.linux-amd64 server \
  --url "##redacted##" \
  --secret "##redacted##" \
  --site "##redacted##" \
  --port 6642 \
  --auth.github.cid "##redacted##" \
  --auth.github.csec "##redacted##" \
  --auth.anon true \
  --image.max-size 0 \
  --smtp.host "smtp.gmail.com" \
  --smtp.port 465 \
  --smtp.tls true \
  --smtp.username "##redacted##" \
  --smtp.password "##redacted##" \
  --auth.email.from "##redacted##@gmail.com" \
  --notify.email.from_address "##redacted##@gmail.com" \
  --notify.users "email" \
  --notify.admins "email" \
  --dbg true
```
The SMTP part works fine, the Email subscription button shows up, entering my email sends the verification code
![image](https://github.com/umputun/remark42/assets/12734857/0e94864f-e21b-4a97-85e9-9851d59d9635)
and pasting it back confirms that I'm now subscribed
![image](https://github.com/umputun/remark42/assets/12734857/50f1a2a9-94ef-4b50-aab8-943b4cfe33c7)

but then, if I test it by adding an anonymous comment (in a private browser window), there's no email sent.

I'm attaching a log (with `--dbg true`) of the following process:
 - starting remark42
 - login via github
 - subscribe to email notifications
 - in a new private window, login as anonymous
 - post a new comment as anonymous
 - (email expected, nothing is sent or seems to be queued)
 - shut down remark
 
Since this is on a site I'm making for a friend, I've search-replaced all domain names etc for their privacy. If anything was lost due to that, please let me know and I'll replicate this setup on my development environment so verbatim logs can be sent.
[remark.log](https://github.com/umputun/remark42/files/11714902/remark.log)

Maybe the issue is some simple misconfiguration? I'd appreciate any tips. Thank you!

**Comment by umputun** [comment, marked as answer]
_pls, use discussions for questions like this. Issues are for the actual errors/bugs. I have transferred the issue_

Is "post a new comment as anonymous" means posting a reply to your comment? Because if not, it won't notify, and docs clearly mention it - "GitHub or Google or Twitter or any other kind of user gets the ability to get email notifications about **new replies to their comments** (and any of the responses down the tree)"

cc: @paskal

**Comment by tjakubo** [comment]
Oh, yes, I absolutely missed that part. Did not even occur to me that might be the case, since the UI for subscribing looks like a standard "subscribe to this thread" feature. I've tested and it works perfectly for notifying me of replies.

Would adding a "subscribe to the whole thread" possibility be something fitting for this project, or discussed and decided against before? I need that feature so would be interested in working towards that on a fork.

Thank you for clearing it up.

**Reply by paskal** [reply]
We thought previously that it wouldn't be an everyday necessity. Also, the complexity of the interface would increase, and you'll need to keep track of a list of pages you receive notifications from and allow users to edit it.

If you want this feature to exist, I propose you create an issue with the description stating what approach you want to take. Historically, the frontend was the most challenging part of the development, so please include that part in your plan for the feature.
